lnwallet/commitment.go

```diff
@@ -225,36 +225,75 @@ func CommitScriptToSelf(chanType channeldb.ChannelType, initiator bool,
 	selfKey, revokeKey *btcec.PublicKey, csvDelay, leaseExpiry uint32) (
 	*ScriptInfo, error) {
 
-	var (
-		toLocalRedeemScript []byte
-		err                 error
-	)
 	switch {
+	// For taproot scripts, we'll need to make a slightly modified script
+	// where the top level key is the revocation case, with our CSV timeout
+	// path living in a tapscript leaf.
+	//
+	// Our "redeem" script here is just the taproot witness program.
+	//
+	// TODO(roasbeef): rework ScriptInfo struct to have taproot specific
+	// info?
+	case chanType.IsTaproot():
```

Comment:
When working on anchor commitments, we had the idea to set up `commitment.go` in a more object-oriented fashion to avoid lots of switches. Not sure how it works out exactly, but could be something to consider for taproot?

Reply:
Hmm, I think in order to do that, we'd break out this switch statement into something like a `ScriptCreator` or `TxCreator` interface, then call the primary method (outputs a script or tx) on that via virtual dispatch. 

We're planning some more aggressive refactors in this areas as prep for integrating taproot assets in a fully external manner (so living in say `litd`). I think at that point we'd take a closer look here, as that path would in theory allow users to define _fully arbitrary_ channel types via lnd config level hooks (so nothing on the RPC, available to other Go programs that embed lnd).